### PR TITLE
test(amazon-cognito-identity-js): coverage for client endpoint option

### PR DESCRIPTION
**Pull request by slaymance**
#### Description of changes

Adds simple unit test coverage for the handling of the `endpoint` option by the `Client` class in the `amazon-cognito-identity-js` flow. ~It also updates the API reference documentation for `AuthClass` to include the `endpoint` option added in [PR 6124](https://github.com/aws-amplify/amplify-js/pull/6124).~

This is a follow-up to [PR 3507](https://github.com/aws-amplify/amplify-js/pull/3507) which originally attempted to add the `endpoint` option to the `Auth` flow.

#### Issue #, if available

[Issue 8244](https://github.com/aws-amplify/amplify-js/issues/8244)

#### Description of how you validated changes
Pull down changes and run `yarn test` to ensure that all tests pass.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/__tests__/Client.test.js [hunk 1]
```diff
@@ -73,5 +73,36 @@ describe('Client unit test suite', () => {
...
+
+			expect(fetch).toHaveBeenCalledWith(
+				`https://cognito-idp.${region}.amazonaws.com/`,
+				expect.any(Object)
+			);
+		});
+
+		test('uses a provided endpoint for requests', async () => {
+			const MOCK_ENDPOINT = 'MOCK_ENDPOINT';
+			const clientWithoutEndpoint = new Client(region, MOCK_ENDPOINT, {});
```
Comment (manueliglesias) [inline packages/amazon-cognito-identity-js/__tests__/Client.test.js:96]:
```suggestion
			const clientWithEndpoint = new Client(region, MOCK_ENDPOINT, {});
```

File: packages/amazon-cognito-identity-js/__tests__/Client.test.js [hunk 2]
```diff
@@ -73,5 +73,36 @@ describe('Client unit test suite', () => {
...
+			jest.spyOn(window, 'fetch');
+			window.fetch.mockResolvedValueOnce({
+				ok: true,
+				json: async () => ({}),
+			});
+
+			await clientWithoutEndpoint.promisifyRequest('', {});
```
Comment (manueliglesias) [inline packages/amazon-cognito-identity-js/__tests__/Client.test.js:103]:
```suggestion
			await clientWithEndpoint.promisifyRequest('', {});
```

**Review by manueliglesias — COMMENTED** [review]
Looking good!

everything inside `docs/api` gets autogenerated, there's no need to include that on the PR

File: docs/api/interfaces/authoptions.html [hunk 3]
```diff
@@ -927,6 +927,9 @@
 			<li class=" tsd-kind-property tsd-parent-kind-interface">
 				<a href="authoptions.html#cookiestorage" class="tsd-kind-icon">cookie<wbr>Storage</a>
 			</li>
+			<li class=" tsd-kind-property tsd-parent-kind-interface">
```
Comment (sammartinez) [inline docs/api/interfaces/authoptions.html:930]: What are these changes from? Did you run a `npm run docs` or a `yarn docs` here?

File: packages/amazon-cognito-identity-js/__tests__/Client.test.js [hunk 4]
```diff
@@ -73,5 +73,36 @@ describe('Client unit test suite', () => {
...
+				ok: true,
+				json: async () => ({}),
+			});
+			jest.fn(window, 'fetch');
+
+			await clientWithoutEndpoint.promisifyRequest('', {});
+
+			expect(fetch).toHaveBeenCalledWith(
+				`https://cognito-idp.${region}.amazonaws.com/`,
+				expect.any(Object)
```
Comment (sammartinez) [inline packages/amazon-cognito-identity-js/__tests__/Client.test.js:90]: Do we know the shape of the Object that will be returned? Or something the customer controls?

File: packages/amazon-cognito-identity-js/__tests__/Client.test.js (hunk 1 above)
Comment (slaymance) [reply packages/amazon-cognito-identity-js/__tests__/Client.test.js:96]: Good catch, I managed to name my constant the opposite of what I wanted 😅

File: docs/api/interfaces/authoptions.html (hunk 3 above)
Comment (slaymance) [reply docs/api/interfaces/authoptions.html:930]: The updates to the API docs were generated using `yarn docs` scoped to `authoptions`. The reason that particular API reference needs to be updated is it lacks any information regarding the `endpoint` option. However, I will remove documentation changes in this PR as Manuel has requested.

File: packages/amazon-cognito-identity-js/__tests__/Client.test.js (hunk 4 above)
Comment (slaymance) [reply packages/amazon-cognito-identity-js/__tests__/Client.test.js:90]: In this case, the shape of the object returned by `fetch` is mocked by the test. The second argument to `fetch` I've opted to expect any `Object`, the rationale being I'm only using these tests to capture the use of the `endpoint` parameter by `Client`. Thoughts? Would this be a better test if mocked the second argument to `fetch` and tested for that as well?

**Review by manueliglesias — APPROVED** [review]
LGTM 🎉

**Pull request merged**